modules/wallet/transactions.go

```diff
@@ -101,18 +116,56 @@ func (w *Wallet) Transactions(startHeight, endHeight types.BlockHeight) (pts []m
 		return nil, errOutOfBounds
 	}
 
-	it := dbProcessedTransactionsIterator(w.dbTx)
-	for it.next() {
-		pt := it.value()
-		if pt.ConfirmationHeight < startHeight {
-			continue
-		} else if pt.ConfirmationHeight > endHeight {
-			// transactions are stored in chronological order, so we can
-			// break as soon as we are above endHeight
-			break
-		} else {
+	// Conduct the binary search
+	var pt modules.ProcessedTransaction
+	cursor := w.dbTx.Bucket(bucketProcessedTransactions).Cursor()
+
+	// Get the largest key present in the processed transactions bucket
+	lastKeyBytes, _ := cursor.Last()
+	if lastKeyBytes == nil {
+		// bucket is empty
+		return
+	}
+
+	// Decode it to the the largest key
+	lastKey := binary.BigEndian.Uint64(lastKeyBytes)
+
+	// Conduct the binary search
+	sort.Search(int(lastKey), func(i int) bool {
+		// Create the key for the index
+		keyBytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(keyBytes, uint64(i))
+
+		// Retrieve the processed transaction
+		key, ptBytes := cursor.Seek(keyBytes)
+		if key == nil {
+			panic("Failed to retrieve processed Transaction by key")
+		}
+
+		// Decode the transaction
+		if success := decodeProcessedTransaction(ptBytes, &pt); !success {
+			panic("Failed to decode the processed transaction")
```

Comment:
I suppose we could also wrap the `sort.Search` call in a function that would catch the panic. I'm not sure how else we could break out of it and return an error.